client/state/selectors/get-media.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Returns media for a specified site ID and query.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Query object
+ * @return {?Array}         Media
+ */
+export default function getMedia( state, siteId, query ) {
+	const queries = state.media.queries[ siteId ];
+
+	if ( ! queries ) {
+		return null;
+	}
+
+	return queries.getItems( query );
```

Comment:
@aduth I'm not familiar with it, so this feedback comes from an ignorant standpoint about the query managers, _caveat emptor_.

> how do you feel about Immutable.js objects in state

this is very different to me than storing functions, though honestly I have a love/hate relationship with `Immutable.js` (or immutable data types in JS in general). different than functions, `Immutable.js` data types are directly serializable and unserializable (I think that it's both proper to use these terms and I think that it's accurate to say that `Immutable.js` data types are isomorphic down to JS objects). we can _compare_ two immutable maps, sets, arrays, etc… whether they are native `Immutable.js` or serialized. We can inspect them and manipulate them.

Functions are inherently _non-comparable_ and _non-serializable_. At best we can save the code for a function as a string and `eval` it on deserialization but it wouldn't take much to demonstrate how bad of an idea that is. The functions are opaque when inspecting the state tree and we can't exactly mock them.

Would it be much of a difficult task to turn `stateObject.getItems( query )` into `getItems( stateObject, query )`?

There must be something I'm missing about the use of `getItems()` and any of its sibling functions if it has any.